viewing_party/party.py

```diff
@@ -0,0 +1,344 @@
+    #     user_watched=[]
+    # friends_watched=[]
+    # friends_unique_watched=[]
+    # for friend in user_data["friends"]:
+    #     for movie in friend["watched"]:
+    #         if movie not in friends_watched:
+    #             friends_watched.append(movie)
+            
+    # if len(user_data["watched"])>0:
+    #     for movie in user_data["watched"]:
+    #         if movie not in user_watched:
+    #             user_watched.append(movie)
+        
+    #     # for friend in user_data["friends"]:
+    #     #     for movie in friend["watched"]:
+    #     #         if movie not in friends_watched:
+    #     #             friends_watched.append(movie)  
+    #     for movie in friends_watched:
+    #         if movie not in user_watched:
+    #             friends_unique_watched.append(movie)
+    #             return friends_unique_watched
+    # return friends_watched
+    
+        
+        # return friends_unique_watched
+
+# *************************************----------------------------*******************************************
+# *************************************----------------------------*******************************************
+# *************************************----------------------------*******************************************
+
+
+def get_available_recs(user_data):
+  
+    recommendations=[]
+    friends_unique_watched=[]
+    
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_unique_watched:
+                friends_unique_watched.append(movie)
+  
+    # why it's not considering the user watched????
+    for movie in friends_unique_watched:
+        # if movie not in user_data["watched"]:
+       
+        if {"title":movie["title"]} not in user_data["watched"]:
+            if movie["host"] in user_data["subscriptions"]:
+                recommendations.append(movie)
+    return recommendations
+   
+
+
+# *************************************----------------------------*******************************************
+# *************************************----------------------------*******************************************
+# *************************************----------------------------*******************************************
+
+def get_new_rec_by_genre(user_data):
+    genre_recommended=[]
+    friends_watched=[]
+    user_watched_genre=[]
+    
+    for movie in user_data["watched"]:
+        user_watched_genre.append(movie["genre"])
+    if len(user_watched_genre)>0:   
+        user_fav_genre=max(set(user_watched_genre), key = user_watched_genre.count)
+
+
+        for friend in user_data["friends"]:
+            for friends_movies_watched in friend["watched"]:
+                if friends_movies_watched not in friends_watched:
+                    friends_watched.append(friends_movies_watched)
+        print(friends_watched)
+        
+        if user_data["watched"] not in friends_watched:
```

Comment:
This will always be True because the types won't match - user_data['watched'] is a list of dictionaries, and friends_watched is a list of dictionaries, so 'in' will compare a list of dictionaries to each dictionary inside of friends_watched.